xesmf/frontend.py

```diff
@@ -105,46 +107,62 @@ def ds_to_ESMFlocstream(ds):
 
     assert lon.shape == lat.shape
 
-    locstream = esmf_locstream(lon, lat)
+    locstream = LocStream.from_xarray(lon, lat)
 
     return locstream, (1,) + lon.shape
 
 
-class Regridder(object):
+def polys_to_ESMFmesh(polys):
+    """
+    Convert a sequence of shapely Polygons to a ESMF.Mesh object.
+
+    MultiPolygons are split in their polygon parts and holes are ignored.
+
+    Parameters
+    ----------
+    polys : sequence of shapely Polygon or MultiPolygon
+
+    Returns
+    -------
+    exterior : ESMF.Mesh
+        A mesh where elements are the exterior rings of the polygons
+    tuple
+        The shape of the mesh : (1, N_elements)
+
+    """
+    ext, holes, _, _ = split_polygons_and_holes(polys)
+    if len(holes) > 0:
+        warnings.warn(
+            'Some passed polygons have holes, those are not represented in the returned Mesh.'
+        )
+    return Mesh.from_polygons(ext), (1, len(ext))
+
+
+class BaseRegridder(object):
     def __init__(
         self,
-        ds_in,
-        ds_out,
+        grid_in,
+        grid_out,
         method,
-        periodic=False,
         filename=None,
         reuse_weights=False,
         extrap_method=None,
         extrap_dist_exponent=None,
         extrap_num_src_pnts=None,
+        add_nans=False,
```

Comment:
no docstring for this option. Is it similar than the fix I introduced in PR #19 ?

Reply:
Woups, I don't exactly remember, but this might be a leftover from my first `BaseRegridder` draft. I'll remove it. NaN's are indeed added to the weights the same way as done in #19, only the conditional statement has changed.